lib/AST/LocalizationFormat.cpp

```diff
@@ -41,8 +46,16 @@ template <> struct ScalarEnumerationTraits<LocalDiagID> {
 
 template <> struct MappingTraits<swift::diag::DiagnosticNode> {
   static void mapping(IO &io, swift::diag::DiagnosticNode &node) {
-    io.mapRequired("id", node.id);
+    // Cast `uint32_t` to `LocalDiagID` to use `diagID` at
```

Comment:
@xedin @LucianoPAlmeida What do you think of these comments, do you think they make sense? If not, any idea how to improve them?

Reply:
If we need "id" to be `LocalDiagID` why not just use a separate variable here (`LocalDiagID diagID;`) to map it and then cast to `uint32_t` to store as `node.id`, I don't see any reason to do this casting...

Also it looks like `DiagnosticNode` itself doesn't have to be a part of `LocalizationFormat.h` and code be moved here instead and placed in anonymous namespace together with `LocalDiagID`.

Reply:
You are actually, right. I just realized that there's actually no reason to use casting here. Because `node.id` doesn't actually have a `DiagID` value, yet. 
The node will get a value after the `io.mapRequired("id", diagID);` gets executed. So, thank you for pointing that out.

I'll also move `DiagnosticNode`.

Reply:
No problem!

Reply:
What about modifying the comment to be 
```   // `diagID` should be of type `LocalDiagID` to use at ScalarEnumerationTraits. ```
Or do you think it's better if it gets removed at all?

Reply:
I don't really think that this clarification is necessary.